File: src/java_caesar/Encryption.java
package java_caesar;

import java.util.Scanner;

public class Encryption {
    private Scanner sc = new Scanner(System.in);
    private String message;
    private String encryptedMessage;
    private int key;

    public Encryption(String message, int key) {
        this.message = message;
        this.key = key;
        this.encryptedMessage = "";
    }

    public Encryption() {
        this.key = 3;
        this.encryptedMessage = "";
    }

    public void plaintextMessage() {
        System.out.print("Enter a message: ");
        message = sc.nextLine();

        System.out.print("Enter key: ");
        key = sc.nextInt();
    }

    public String getEncryptedMessage() {
        return encryptedMessage;
    }

    public String getMessage() {
        return message;
    }

    public int getKey() {
        return key;
    }

    public void encrypt() {
        for(int i = 0; i < message.length(); ++i){
            char ch = message.charAt(i);

            if(ch >= 'a' && ch <= 'z'){
                ch = (char)(ch + key);

                if(ch > 'z'){
                    ch = (char)(ch - 'z' + 'a' - 1);
                }

                encryptedMessage += ch;
            }
            else if(ch >= 'A' && ch <= 'Z'){
                ch = (char)(ch + key);

                if(ch > 'Z'){
                    ch = (char)(ch - 'Z' + 'A' - 1);
                }

                encryptedMessage += ch;
            }
            else {
                encryptedMessage += ch;
            }
        }

        System.out.println("Encrypted Message = " + encryptedMessage);
    }
}

File: src/kotlin_caesar/Encryption.kt
package kotlin_caesar

import java.util.*

class Encryption {
    private val sc: Scanner = Scanner(System.`in`)
    private var message: String? = null
        private set
    var encryptedMessage: String
        private set
    var key: Int
        private set

    constructor(message: String?, key: Int) {
        this.message = message
        this.key = key
        encryptedMessage = ""
    }

    constructor() {
        key = 3
        encryptedMessage = ""
    }

    fun plaintextMessage() {
        print("Enter a message: ")
        message = sc.nextLine()
        print("Enter key: ")
        key = sc.nextInt()
    }

    /**
     * The 'if' loop is replaced by the 'when' loop.
     */
    fun encrypt() {
        for (element in message!!) {
            var ch = element
            // below the commented 'if' version of the loop
            when (ch) {
                in 'a'..'z' -> {
                    ch = (ch.toInt() + key).toChar()
                    if (ch > 'z') {
                        ch = (ch - 'z' + 'a'.toInt() - 1).toChar()
                    }
                    encryptedMessage += ch
                }
                in 'A'..'Z' -> {
                    ch = (ch.toInt() + key).toChar()
                    if (ch > 'Z') {
                        ch = (ch - 'Z' + 'A'.toInt() - 1).toChar()
                    }
                    encryptedMessage += ch
                }
                else -> {
                    encryptedMessage += ch
                }
            }
        }
        println("Encrypted Message = $encryptedMessage")
    }
}

/*
if (ch in 'a'..'z') {
    ch = (ch.toInt() + key).toChar()
    if (ch > 'z') {
        ch = (ch - 'z' + 'a'.toInt() - 1).toChar()
    }
    encryptedMessage += ch
} else if (ch in 'A'..'Z') {
    ch = (ch.toInt() + key).toChar()
    if (ch > 'Z') {
        ch = (ch - 'Z' + 'A'.toInt() - 1).toChar()
    }
    encryptedMessage += ch
} else {
    encryptedMessage += ch
}*/